### PR TITLE
Add option to bail out of `jake runtests` when a test fails

### PR DESCRIPTION
**Pull request by andy-ms**
Currently serial only. For parallel we would need a way to notify all other running processes to stop.

### CODE REVIEW

**Comment by sandersn** [comment]
:+1:

For parallel runs, `finishWorker` used to call `fail` for all cases when `errorStatus !== undefined`. Now it only does so for the last worker. It would be easy to restore that `fail` call behind a check on `--bail`.

File: Jakefile.js [hunk 1]
```diff
@@ -755,13 +755,18 @@ function runConsoleTests(defaultReporter, runInParallel) {
...
+                var workerCmd = "mocha" + " -t " + testTimeout + " -R " + reporter + " " + colors + bail + " " + run + " --config='" + configPath + "'";
                 console.log(workerCmd);
                 exec(workerCmd,  finishWorker, finishWorker) 
             });
             
             function finishWorker(e, errorStatus) {
                 counter--;
+
+                if (bail && errorStatus !== undefined) {
+                    failWithStatus(firstErrorStatus);
```
Comment (sandersn) [inline Jakefile.js]: Does this fail with the correct number of errors? I don't think `firstErrorStatus` would be assigned until after the next `if` block.

I guess the fix would be to move below the next `if` block or just return `errorStatus`

**Comment by sandersn** [comment]
I commented on the commit by mistake so it doesn't show up on this page.

**Comment by sandersn** [comment]
Do you want to just drop the parallel commit? The parallel code is now very complex and it runs in 30-60 seconds, so there's not much point in bailing anyway.

**Comment by andy-ms** [comment]
Good idea. Replacing with #9146

**Pull request closed without merging**